Core/GDCore/Project/EventsFunctionsExtension.cpp

```diff
@@ -185,11 +189,93 @@ void EventsFunctionsExtension::UnserializeExtensionImplementationFrom(
+  // before they are unserialized.
+
+  // At the beginning, everything is yet to be loaded.
+  std::vector<gd::String> remainingEventsBasedObjectNames(
+      eventsBasedObjects.size());
+  for (std::size_t i = 0; i < eventsBasedObjects.size(); ++i) {
+    remainingEventsBasedObjectNames[i] = eventsBasedObjects.at(i).GetName();
+  }
+
+  // Helper allowing to find if an object depends on at least one other object from
+  // the extension that is not loaded yet.
+  auto &extensionName = name;
+  auto isDependentFromRemainingEventsBasedObjects =
+      [&remainingEventsBasedObjectNames,
+       &extensionName](const gd::SerializerElement &eventsBasedObjectElement) {
+        auto &objectsElement = eventsBasedObjectElement.GetChild("objects");
+        objectsElement.ConsiderAsArrayOf("object");
+
+        for (std::size_t objectIndex = 0;
+             objectIndex < objectsElement.GetChildrenCount(); ++objectIndex) {
+          const gd::String &objectType =
+              objectsElement.GetChild(objectIndex).GetStringAttribute("type");
+
+          gd::String usedExtensionName =
+              PlatformExtension::GetExtensionFromFullObjectType(objectType);
+          if (usedExtensionName != extensionName) {
+            // The object comes from another extension: the project is already responsible
+            // for loading extensions in the proper order.
+            continue;
+          }
+          gd::String eventsBasedObjectName =
+              gd::PlatformExtension::GetObjectNameFromFullObjectType(
+                  objectType);
+
+          if (std::find(remainingEventsBasedObjectNames.begin(),
+                        remainingEventsBasedObjectNames.end(),
+                        eventsBasedObjectName) !=
+              remainingEventsBasedObjectNames.end()) {
+            return true;
+          }
+        }
+        return false;
+      };
+
+  // Find the order of loading so that the objects are loaded when all the objects
+  // they depend on are already loaded.
+  std::vector<gd::String> loadOrderEventsBasedObjectNames;
+  bool foundAnyEventsBasedObject = true;
+  while (foundAnyEventsBasedObject) {
+    foundAnyEventsBasedObject = false;
+    for (std::size_t i = 0; i < remainingEventsBasedObjectNames.size(); ++i) {
+      auto eventsBasedObjectName = remainingEventsBasedObjectNames[i];
+      size_t extensionIndex = eventsBasedObjects.GetPosition(
+          eventsBasedObjects.Get(eventsBasedObjectName));
+      const SerializerElement &eventsBasedObjectElement =
+          eventsBasedObjectsElement.GetChild(extensionIndex);
+
+      if (!isDependentFromRemainingEventsBasedObjects(
+              eventsBasedObjectElement)) {
+        loadOrderEventsBasedObjectNames.push_back(eventsBasedObjectName);
+        remainingEventsBasedObjectNames.erase(
+            remainingEventsBasedObjectNames.begin() + i);
+        i--;
+        foundAnyEventsBasedObject = true;
+      }
+    }
```

Comment:
I think an object A depending on B and B depending on A will result in an infinite loop?

Reply:
The editor doesn't allow it, but they would only be ignored and not loaded which is not the best but still fine I guess.
We could log what remains in `remainingEventsBasedObjectNames` at the end of the loop.